core/dbt/rpc.py

```diff
@@ -130,6 +131,54 @@ def __getitem__(self, key):
         return self.rpc_factory(task)
 
 
+def _nt_setup(config, args):
+    """
+    On windows, we have to do a some things that dbt does dynamically at
+    process load.
+
+    These things are inherited automatically on posix, where fork() keeps
+    everything in memory.
+    """
+    # reload the active plugin
+    load_plugin(config.credentials.type)
+
+    # reset tracking, etc
+    config.config.set_values(args.profiles_dir)
+
+
+def _task_bootstrap(task, queue, kwargs):
+    signal.signal(signal.SIGTERM, sigterm_handler)
+    # the first thing we do in a new process: start logging
+    add_queue_handler(queue)
+    # on windows, we need to reload our plugins because of how it starts new
+    # processes. At this point there are no adapter plugins loaded!
+    if os.name == 'nt':
+        _nt_setup(task.config, task.args)
```

Comment:
which types of values are unset after the process is spawned? All global variables? Do we need to make considerations for `flags` here?

Reply:
Oooh, that's a great point! We definitely do!

The values that are unset are any values that are changed by runtime behavior of the program. Essentially, in Python on Windows, all source files are re-parsed as if they were being loaded from the first time at module start within the new process, and then python executes whatever function + args you've passed the Process object.